hilti/toolchain/include/compiler/detail/cxx/unit.h

```diff
@@ -36,7 +37,7 @@ class Linker;
 
 namespace linker {
 
-using MetaData = nlohmann::json;
+using MetaData = hilti::rt::ValueReference<nlohmann::json>;
```

Comment:
We don't seem to need the shared behavior of `ValueReference`, so maybe we should use something like `std::unique_ptr` to make this work with fwd decls instead (makes construction slightly more ugly in an isolated number of places).

Reply:
I wanted to keep the value semantics, not immediately sure if changing that would break anything. Sounds like you're saying it wouldn't?

Reply:
I didn't check, but I would expect that should mostly work as we don't seem to store `Metadata`. We'll probably would need fix function returns to use `std::move` though.

Reply:
So I started to make that change, but turns out it's completely trivial at least due to current ownership semantics (or lack thereof).Will just leave as is for now.